sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1405,4 +1426,35 @@ class AnsiCastSuite extends CastSuiteBase {
       checkEvaluation(cast(negativeTs, LongType), expectedSecs)
     }
   }
+
+  test("Fast fail for cast string type to decimal type in ansi mode") {
+    checkEvaluation(cast("12345678901234567890123456789012345678", DecimalType(38, 0)),
+      Decimal("12345678901234567890123456789012345678"))
```

Comment:
Is this behaivour different between ANSI enabled/disabled? I think this test should only include specific behaivours in the ANSI mode.

Reply:
Yes. It should throw exception In ANSI mode.